pkg/helper/autonode/helpers.go

```diff
@@ -0,0 +1,156 @@
+package autonode
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+// SetAutoNode validates and sets AutoNode configuration for a cluster
+func SetAutoNode(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster, autonode string, roleArn string) (
+	autonodeMode string, autonodeRole string, err error) {
+
+	// Check if flags are changed
+	autonodeChanged := cmd.Flags().Changed("autonode")
+	roleArnChanged := cmd.Flags().Changed("autonode-iam-role-arn")
+
+	if !autonodeChanged && !roleArnChanged {
+		return "", "", nil
+	}
+
+	// Validate HCP cluster
+	if !aws.IsHostedCP(cluster) {
+		return "", "", fmt.Errorf("AutoNode is only supported for Hosted Control Plane clusters")
+	}
+
+	currentAutonodeEnabled := ocm.AutoNodeExists(cluster)
+
+	// Validate autonode value if provided
+	if autonodeChanged {
+		if autonode != "enabled" {
+			return "", "", fmt.Errorf("Invalid value for --autonode. Currently only 'enabled' is supported")
+		}
+
+		// Check if trying to enable when already enabled
+		if currentAutonodeEnabled {
+			return "", "", fmt.Errorf("AutoNode is already enabled for this cluster")
+		}
+
+		// When enabling AutoNode, IAM role ARN is required
+		if !roleArnChanged || roleArn == "" {
+			return "", "", fmt.Errorf("IAM role ARN is required when enabling AutoNode")
+		}
+
+		autonodeMode = autonode
+	}
+
+	// Validate IAM role ARN if provided
+	if roleArnChanged {
+		if roleArn == "" {
+			return "", "", fmt.Errorf("IAM role ARN cannot be empty")
+		}
+
+		// Validate ARN format
+		if !aws.RoleArnRE.MatchString(roleArn) {
+			return "", "", fmt.Errorf("Invalid IAM role ARN format: '%s'. Expected format: arn:aws:iam::<account-id>:role/<role-name>", roleArn)
+		}
+
+		// If AutoNode is not being enabled in this request, it must already be enabled
+		if !autonodeChanged && !currentAutonodeEnabled {
+			return "", "", fmt.Errorf("Cannot update IAM role ARN when AutoNode is not enabled. " +
+				"Enable AutoNode first with --autonode=enabled")
+		}
+
+		// If AutoNode is already enabled and only updating the role
+		if currentAutonodeEnabled && !autonodeChanged {
+			autonodeMode = "enabled" // Maintain the current state
+		}
+
+		autonodeRole = roleArn
+	}
+
+	// Display confirmation prompts based on the changes being made
+	if autonodeMode == "enabled" && !currentAutonodeEnabled {
+		// Enabling AutoNode for the first time
+		r.Reporter.Warnf("You are choosing to enable automatic node management (AutoNode)")
+		if !confirm.Confirm("enable AutoNode for cluster with the provided role arn '%s'", autonodeRole) {
```

Comment:
is that role arn the cluster's role arn? Or the auto node role arn?

Reply:
@hunterkepley It is the AutoNode role ARN.